sdk/ml/azure-ai-ml/tests/dsl/unittests/test_io_builder.py

```diff
@@ -34,3 +43,118 @@ def test_nested_input_output_builder(self):
+                'limits': None,
+                'name': 'node2',
+                'outputs': {'component_out_path': {'type': 'literal',
+                                                   'value': '${{parent.outputs.output2}}'}},
+                'resources': None,
+                'tags': {},
+                'type': 'command'}
+        }
+        assert rest_pipeline_job["jobs"] == expected_pipeline_job1
+
+        pipeline_job2 = my_pipeline(
+            job_in_number=2, job_in_path=Input(path="fake_path2")
+        )
+
+        rest_pipeline_job = omit_with_wildcard(pipeline_job2._to_rest_object().properties.as_dict(),
+                                               *common_omit_fields)
+
+        expected_pipeline_job2 = {
+            'node1': {'computeId': None,
+                      'display_name': None,
+                      'distribution': None,
+                      'environment_variables': {},
+                      'inputs': {'component_in_number': {'job_input_type': 'literal',
+                                                         'value': '${{parent.inputs.job_in_number}}'},
+                                 'component_in_path': {'job_input_type': 'literal',
+                                                       'value': '${{parent.inputs.job_in_path}}'}},
+                      'limits': None,
+                      'name': 'node1',
+                      'outputs': {'component_out_path': {'type': 'literal',
+                                                         'value': '${{parent.outputs.output1}}'}},
+                      'resources': None,
+                      'tags': {},
+                      'type': 'command'},
+            'node2': {'computeId': None,
+                      'display_name': None,
+                      'distribution': None,
+                      'environment_variables': {},
+                      'inputs': {'component_in_number': {'job_input_type': 'literal',
+                                                         'value': '2'},
+                                 'component_in_path': {'job_input_type': 'uri_folder',
+                                                       'uri': 'fake_path2'}},
+                      'limits': None,
+                      'name': 'node2',
+                      'outputs': {'component_out_path': {'type': 'literal',
+                                                         'value': '${{parent.outputs.output2}}'}},
+                      'resources': None,
+                      'tags': {},
+                      'type': 'command'}
+        }
+        assert rest_pipeline_job["jobs"] == expected_pipeline_job2
+
+        # calling pipeline func again won't affect existing pipeline job
+        pipeline_job1.jobs["node2"].inputs["component_in_number"] == 1
+        pipeline_job1.jobs["node2"].inputs["component_in_path"].path == "fake_path1"
+
+        rest_pipeline_job = omit_with_wildcard(pipeline_job1._to_rest_object().properties.as_dict(),
```

Comment:
add test case that we have multi layer of subgraph

Reply:
Added test